### PR TITLE
Update section on path migration

### PR DESCRIPTION
**Pull request by mirjak**
addresses #115 (partially at least)

### CODE REVIEW

File: draft-ietf-quic-applicability.md [hunk 1]
```diff
@@ -451,16 +451,21 @@ firewalls that rely on the port number for application identification.
 
 # Connection Migration
 
-QUIC supports connection migration. Even if lower-layer addresses (usually the
-4-tuple of IP addresses and ports) changes, QUIC packets can still be associated
-with an existing connection based on the Connection ID (see also section
-{{connid}}) in the QUIC header, if present. This supports cases where address
-information changes due to e.g. NAT rebinding or change of the local interface.
+QUIC supports connection migration by the client. If a lower-layer address
+changes, a QUIC endpont can still associate packets with an existing connection
```
Comment (britram) [inline draft-ietf-quic-applicability.md:455]:
```suggestion
changes, a QUIC endpoint can still associate packets with an existing connection
```

File: draft-ietf-quic-applicability.md [hunk 2]
```diff
@@ -451,16 +451,21 @@ firewalls that rely on the port number for application identification.
...
+if present. This supports cases where address information changes due to e.g.
+NAT rebinding, intentional change of the local interface, or based on an
+indication in the handshake of the server for a preferred address to be used. 
+
 Currently QUIC only supports failover cases. Only one "path" can be used at a
 time, and as soon as the new path is validated all traffic will be switched over
-to the next path. Of course if an endpoint decided to not use the Connection ID
-in short packets (Zero-length Conn ID) for a certain connection, migration is
-not supported for that direction of the connection.
+to the next path. For intention migrations the client can send probing packets
```
Comment (britram) [inline draft-ietf-quic-applicability.md:463]:
```suggestion
to the next path. For intentional migrations the client can send probing packets
```

File: draft-ietf-quic-applicability.md [hunk 3]
```diff
@@ -451,16 +451,21 @@ firewalls that rely on the port number for application identification.
...
+before switching over to measurement and eventually pre-heat the path. The other
+endpoint in required to validate the new path before use in order to avoid
+address spoofiing attacks. Path validation takes one RTT and congestion control
+will also be reset on path mitragtion. Therefore migration usual has a performance
```
Comment (britram) [inline draft-ietf-quic-applicability.md:467]:
```suggestion
will also be reset on path miragtion. Therefore migration usually has a performance
```

**Review by britram — COMMENTED** [review]
if we want to talk about preheating, then this looks good...

File: draft-ietf-quic-applicability.md [hunk 4]
```diff
@@ -451,16 +451,21 @@ firewalls that rely on the port number for application identification.
...
 # Connection Migration
 
-QUIC supports connection migration. Even if lower-layer addresses (usually the
-4-tuple of IP addresses and ports) changes, QUIC packets can still be associated
-with an existing connection based on the Connection ID (see also section
-{{connid}}) in the QUIC header, if present. This supports cases where address
-information changes due to e.g. NAT rebinding or change of the local interface.
+QUIC supports connection migration by the client. If a lower-layer address
+changes, a QUIC endpoint can still associate packets with an existing connection
+based on the Connection ID (see also section {{connid}}) in the QUIC header,
```
Comment (martinthomson) [inline draft-ietf-quic-applicability.md:456]:
```suggestion
based on the Connection ID (see also {{connid}}) in the QUIC header,
```

File: draft-ietf-quic-applicability.md [hunk 5]
```diff
@@ -451,16 +451,21 @@ firewalls that rely on the port number for application identification.
...
+if present. This supports cases where address information changes due to e.g.
```
Comment (martinthomson) [inline draft-ietf-quic-applicability.md:457]: Avoid "e.g." used in this way.  Try to say things like "such as" instead.

File: draft-ietf-quic-applicability.md [hunk 6]
```diff
@@ -451,16 +451,21 @@ firewalls that rely on the port number for application identification.
...
+NAT rebinding, intentional change of the local interface, or based on an
+indication in the handshake of the server for a preferred address to be used.
+
 Currently QUIC only supports failover cases. Only one "path" can be used at a
 time, and as soon as the new path is validated all traffic will be switched over
-to the next path. Of course if an endpoint decided to not use the Connection ID
-in short packets (Zero-length Conn ID) for a certain connection, migration is
-not supported for that direction of the connection.
+to the next path. For intentional migrations the client can send probing packets
+before switching over to measure the path characteristics and eventually pre-heat
```
Comment (martinthomson) [inline draft-ietf-quic-applicability.md:464]: "pre-heat" is a colloquialism that should be avoided.  Note that probing does very little to allow the congestion window to open, so this is questionable anyway.

File: draft-ietf-quic-applicability.md [hunk 7]
```diff
@@ -451,16 +451,21 @@ firewalls that rely on the port number for application identification.
...
+the congestion controller. The other endpoint in required to validate the new path
+before use in order to avoid address spoofiing attacks. Path validation takes one
```
Comment (martinthomson) [inline draft-ietf-quic-applicability.md:466]:
```suggestion
before use in order to avoid address spoofing attacks. Path validation takes one
```

File: draft-ietf-quic-applicability.md [hunk 8]
```diff
@@ -451,16 +451,21 @@ firewalls that rely on the port number for application identification.
...
+RTT and congestion control will also be reset on path miragtion. Therefore
```
Comment (martinthomson) [inline draft-ietf-quic-applicability.md:467]:
```suggestion
RTT and congestion control will also be reset on path migration. Therefore
```

File: draft-ietf-quic-applicability.md (hunk 7 above)
Comment (martinthomson) [reply draft-ietf-quic-applicability.md:466]: And maybe:

```suggestion
before use in order to avoid address spoofing attacks. Path validation takes at least one
```

File: draft-ietf-quic-applicability.md [hunk 9]
```diff
@@ -513,7 +518,8 @@ Connection ID.
 Note that the Connection ID in the short header may be omitted. This is a
 per-connection configuration option; if the Connection ID is not present, then
 the peer omitting the connection ID will use the same local address for the
```
Comment (martinthomson) [inline draft-ietf-quic-applicability.md:520]: "will" or "needs to"?

File: draft-ietf-quic-applicability.md (hunk 6 above)
Comment (mirjak) [reply draft-ietf-quic-applicability.md:464]: We discussed this ones in a multipath related discussion. The idea here was to send multiple probing packets (and II guess padding). If that is seen as a viable approach for e.g. mobile handovers, it would be good to mention is somewhere. However, I guess we would then need to say more. I believe @erickinnear brought this up... otherwise if people think it's too early to say anything like that,  can also remove the half sentence.

File: draft-ietf-quic-applicability.md (hunk 9 above)
Comment (britram) [inline draft-ietf-quic-applicability.md:520]:
```suggestion
the peer omitting the connection ID needs to use the same local address for the
```

File: draft-ietf-quic-applicability.md (hunk 5 above)
Comment (britram) [reply draft-ietf-quic-applicability.md:457]:
```suggestion
if present. This supports cases where address information changes, such as
```

**Review by britram — APPROVED** [review]

File: draft-ietf-quic-applicability.md (hunk 6 above)
Comment (gorryfair) [reply draft-ietf-quic-applicability.md]: I'm in favour of not saying anything after the /and/.

Comment (mirjak) [reply draft-ietf-quic-applicability.md]: @gorryfair this PR was merged a long time ago. let's have the discussion in issue #197. As you can see that issue is flagged for meeting discussion.
